.adr/0017-missed-chain-events.md

```diff
@@ -0,0 +1,75 @@
+# 0017 Missed Chain Events
+
+## Status
+
+Review
+
+## Context
+
+A user should be able to restart a nitro node without concern that it has missed chain events that caused the node's data to get out of sync with other nodes and the chain state. Similarly, a nitro node should be able to start for the first time and catch up on any relevant chain events that occurred before the node started. To address these concerns, a nitro node should check for any chain events it might have missed since it was last online, and queue those events to be processed as part of the node's initialization routines.
+
+If the `chainservice` communicates with the `engine` when new blocks are processed, the node can store the `lastBlockSeen` in the `durablestore`. Then, when the node is initialized it can read the `lastBlockSeen` from the `store`, and search for any chain events that occurred between that block and the current block.
```

Comment:
Yes the `memstore` just checks from genesis every time when restarting

Reply:
Ok, perhaps let us put that in the ADR then. Does the `engine` need to know about what kind of store it is dealing with? It only has an `interface` right?

Reply:
That's correct, the `engine` will just call the `store`'s interface, regardless of which `store` type is being used. During the `InitializeNode` function, we call `store.GetLastBlockNumSeen()`. For the `memstore` this will always return `0`. https://github.com/statechannels/go-nitro/blob/main/internal/node/node.go#L36

Reply:
Added sentence to explain `memstore` init: https://github.com/statechannels/go-nitro/pull/1624/commits/201969ba1cf3f96300a2affdfcd4bcd0b0ea2d61

Reply:
Thanks!